File: src/main/java/com/github/lppedd/cc/documentation/CommitTypeDocumentationProvider.java
package com.github.lppedd.cc.documentation;

import org.jetbrains.annotations.Nullable;

import com.github.lppedd.cc.psi.CommitTypePsiElement;
import com.intellij.lang.documentation.AbstractDocumentationProvider;
import com.intellij.psi.PsiElement;

public class CommitTypeDocumentationProvider extends AbstractDocumentationProvider {
  @Nullable
  @Override
  public String generateDoc(
      final PsiElement element,
      @Nullable final PsiElement originalElement) {
    if (!(element instanceof CommitTypePsiElement)) {
      return null;
    }

    final CommitTypePsiElement commitType = (CommitTypePsiElement) element;
    return commitType.getCommitTypeDescription();
  }
}

File: src/main/kotlin/com/github/lppedd/cc/documentation/CommitTypeDocumentationProvider.kt
package com.github.lppedd.cc.documentation

import com.github.lppedd.cc.lookupElement.CommitLookupElement
import com.github.lppedd.cc.psi.CommitTypePsiElement
import com.intellij.lang.documentation.AbstractDocumentationProvider
import com.intellij.psi.PsiElement
import com.intellij.psi.PsiManager

/**
 * @author Edoardo Luppi
 */
internal class CommitTypeDocumentationProvider : AbstractDocumentationProvider() {
  override fun generateDoc(element: PsiElement, originalElement: PsiElement?) =
    if (element is CommitTypePsiElement) {
      val description = element.commitType.description ?: ""
      if (!description.isBlank()) description else null
    } else {
      null
    }

  override fun getDocumentationElementForLookupItem(
    psiManager: PsiManager?,
    obj: Any?,
    element: PsiElement?
  ) = if (obj is CommitLookupElement) {
    obj.psiElement
  } else {
    super.getDocumentationElementForLookupItem(psiManager, obj, element)
  }
}
